src/connector/mysql/error.rs

```diff
@@ -10,6 +10,9 @@ impl From<my::Error> for Error {
                 message: err.to_string(),
             })
             .build(),
+            my::Error::Io(my::IoError::Io(err)) if err.kind() == std::io::ErrorKind::UnexpectedEof => {
```

Comment:
weird, I wouldn't have identified these two kinds of errors at first sight, but it sort of makes sense